qiita_db/metadata_template/prep_template.py

```diff
@@ -289,6 +289,45 @@ def data_type(self, ret_id=False):
             "p.prep_template_id=%s".format(ret), (self.id,))[0]
 
     @property
+    def columns_restrictions(self):
+        """Gets the dictionary of colums required based on data_type
+
+        Returns
+        -------
+        dict
+            The dict of restictions based on the data_type
+        """
+        pt_cols = deepcopy(PREP_TEMPLATE_COLUMNS)
+        if self.data_type() in TARGET_GENE_DATA_TYPES:
+            pt_cols.update(PREP_TEMPLATE_COLUMNS_TARGET_GENE)
+
+        return pt_cols
+
+    def can_be_updated(self, columns):
+        """Gets if the template can be updated
+
+        Parameters
+        ----------
+        columns : set
+            A set of the names of the columns to be updated
+
+        Returns
+        -------
+        bool
+            If the template can be updated
+        """
+        if not self.preprocessed_data:
+            return True
+
+        tg_columns = {e for v in viewvalues(PREP_TEMPLATE_COLUMNS_TARGET_GENE)
```

Comment:
I think this test should only be performed if the prep template is of a TARGET GENE type.
I'm unsure if other data types will have different requirements at this level, so it is better to not check this always.